sklearn/utils/tests/test_stats.py

```diff
@@ -96,3 +96,71 @@ def test_weighted_percentile_2d():
         _weighted_percentile(x_2d[:, i], w_2d[:, i]) for i in range(x_2d.shape[1])
     ]
     assert_allclose(w_median, p_axis_0)
+
+
+def test_weighted_percentile_nan_filtered():
+    """Test that calling _weighted_percentile on an array with nan values returns
+    the same results as calling _weighted_percentile on a filtered version of the data.
+    We test both with sample_weight of the same shape as the data and for
+    one-dimensional sample_weight."""
+
+    rng = np.random.RandomState(42)
+    array_with_nans = rng.rand(10, 100)
+    array_with_nans[rng.rand(*array_with_nans.shape) < 0.5] = np.nan
+    nan_mask = np.isnan(array_with_nans)
+
+    sample_weights = [rng.randint(1, 6, size=(10, 100)), rng.randint(1, 6, size=(10,))]
+
+    for sample_weight in sample_weights:
```

Comment:
Instead of doing a loop inside the test, we could parametrize this test as follows:

```python
@pytest.mark.parametrize("sample_weight_ndim": [1, 2])
def test_weighted_percentile_nan_filtered(sample_weight_ndim):
    ...
        
    if sample_weight_ndim == 2:
        sample_weight = rng.randint(1, 6, size=(10, 100)
    else:
        sample_weight = rng.randint(1, 6, size=(10,))
```

This way, if the test starts to fail only for one of the two settings, the failure report will be more explicit.

Reply:
Done. 

It seems that `rng` is never used outside of the testing functions throughout the project so it cannot be used to parametrize the testing functions like this `@pytest.mark.parametrize("sample_weight", [rng.randint(1, 6, size=(10, 100)), rng.randint(1, 6, size=(10,))])`.